### PR TITLE
[Docs] Separating documentation in multiple files, and make main README with less information

### PR DESCRIPTION
**Pull request by Nipodemos**
This is a draft, it should not be merged yet.

Hello @jonataslaw !
I am thinking about this for a few days, but i don't even know if that is a good idea. Since README is getting gigantic and it will be even bigger when everything gets documented, the main readme file is getting bloated with information. It's a lot for one place.

My proposal is to split documentation in different files, make the main README have just a resumed version of all GetX can do, and link for the more extended documentation.

Advantages:
* The first contact of new users with get can be better, improving the amount of people that want to try it out, and making easier for them to do this
* It is easier to maintain, because, if you make an update in just one thing, you can change just one file
* more organization in docs

Disavantages:
* Since it will not have all features on one file, the user can't simply use CTRL + F and search the term he/she wants to learn more
* We will have to focus on two documentations: the resumed version and the extended version

I think it worth it.
If you like the idea, can you give me some advice of what to put in the main README? I'm with little creativity hahah

### CODE REVIEW

**Comment by jonataslaw** [comment]
I like the idea. I hope this doesn't cause any more issues because the person hasn't "seen" the other readmes.
This is also a concern of mine, the readme has become one of the largest library readmes within git, it is gigantic, maybe the readme is just for initial instructions and I am using it as "documentation".
I think the essentials are the navigation methods, the examples of the state managers, and the examples of the instance manager, but even so, I think that perhaps the readme is still at a limit, everything that goes beyond what is written is exaggeration , and translations, GetUtils and Queue are still missing.

**Comment by stefandevo** [comment]
Can't you use the WIKI portion of github here to split the files?

**Comment by Nipodemos** [comment]
i'll put the link both in beginning and end of the topic. Maybe it helps.

I've put the three main parts of GetX under the name "The three pillars". I could not think of a better name on that time, do you have a better idea?

I didn't understand what you meant with this part: `but even so, I think that perhaps the readme is still at a limit, everything that goes beyond what is written is exaggeration`. Could you rephrase it?

i've noticed you put a bunch of methods on GetUtils and some translation stuff and  didn't said a word about it hahahaha, you must've be very busy. Now, i have no ideia what is this Queue you are talking about.

tomorrow i'll work more on this PR, now i will sleep

**Comment by Nipodemos** [comment]
> Can't you use the WIKI portion of github here to split the files?

@stefandevo actually, that is a good idea! But i don't know how to use this wiki... Do we even have permission to edit? or should we send all of our changes to @jonataslaw to only them he puts there?

well, if anyone can edit, then it will be a lot better to use wiki.

**Comment by stefandevo** [comment]
@Nipodemos some info:
https://help.github.com/en/github/building-a-strong-community/changing-access-permissions-for-wikis

So you give acces to contributors for example to edit the wiki. But of course @jonataslaw needs to set it up (and trust the contributors :-) )

**Comment by Nipodemos** [comment]
@stefandevo i like the idea, if @jonataslaw likes then we can do this
for now i'll update this PR, because when we migrate to wiki, it will be easy as copy & paste 😄 

@jonataslaw i put a `why i made this package` topic in documentation, putting your words of an issue i read. Hope you like it 😄 
I think it is ready for review, since this PR is not to put new documentation, it is just to organize

**Comment by Nipodemos** [comment]
i tried to mess with github wiki a bit, but i've noticed a problem:
**it is harder for people to contribute**

When a wiki is made, only people that have permission to edit can edit. That is necessary for security/safety reasons, but a person who wants to improve or to make a new page (like for example, that guy that translated the whole readme to spanish) can't do that. They can't make a pull request on wiki either. Here are the options:
1. He opens a issue asking for permission on the repository, but jonatas can't give permission to anyone, since you can do anything, even push code directly.
2. Put his changes on a gist or something, create an issue warning @jonataslaw about his desire to improve docs, then jonatas would get his gist and make manually a new commit. 
3. Send the file via email to jonatas and ask him to make a new commit on wiki

It Is just getting harder and harder to help! 

For that reason, i think we should give up Github Wiki, and think on other thing. One that supports multi language, and docs versioning

**Comment by jonataslaw** [comment]
I think it is only possible to disable the Wiki now, not to delete it.

Maybe doing something external like Kauê is doing here https://kauemurakami.github.io/getx_pattern/ (Kauê is the one who created the snippet for vscode) would be a good thing.
Difficult is that generally doing this type of thing would involve working with html, something that not everyone is used to.
Yes, this is a situation that we will have to analyze how to resolve.

**Comment by stefandevo** [comment]
I think you can use github pages. It is just a repo and then people can contribute.

**Comment by jonataslaw** [comment]
I saw now that github generated a page from the Readme's markdown, and updates it whenever a change is made.
There is the possibility to insert another branch, and select for it, in addition to the possibility of creating a new repository just for that.
It might be a good idea to do this.
https://jonataslaw.github.io/getx/

**Comment by jonataslaw** [comment]
There is an option on the wiki to open for everyone to edit, I have temporarily enabled it now for testing, but I don't know if I need to authorize it, like a merge, or something.

**Comment by Nipodemos** [comment]
i liked the idea of github pages, now we just need to find a option to have a sidebar with every file of documentation, i wonder if they have that already set up

**Comment by Nipodemos** [comment]
yeah, github pages doesn't offer the option of different languages, but that's what we get for now
i'll update the other languages to use the same layout as the english version

**Comment by jonataslaw** [comment]
I have a domain purchased from getx.site that I used to validate the package on pub.dev, and some servers stopped (I pay some servers annually to do tests) I am thinking of doing the Get documentation, using Get with Flutter web, and put on that site. Perhaps it is the best option of all. More laborious, to be sure, but certainly the best option of all.

**Comment by Nipodemos** [comment]
@jonataslaw i totally agree, and i think that would be an awesome experience. But recently i am with little free time, studying a lot. For now i can change these files, and in a short future, the flutter web site can be made

**Comment by jonataslaw** [comment]
We can use Get's internationalization to change the language, and dynamic routes to access content directly via the link.
The language can be saved using GetStorage so that if the person visits it again, they can read everything in their preferred language. A little ambitious, but I think the right thing to do.
I would like to have a better communication channel too, currently there are groups spread across the telegram, discord, gitter, Facebook, but nothing officially supported.

**Comment by jonataslaw** [comment]
> @jonataslaw i totally agree, and i think that would be an awesome experience. But recently i am with little free time, studying a lot. For now i can change these files, and in a short future, the flutter web site can be made

I know that, we all have time limitations. So I think about making the "skeleton", and the text part can be inserted lazily.
This done in a repository will probably attract more collaborators, since I see that there are a lot of people who are afraid to offer PRs and something is wrong. Those people who want to contribute, but do not feel at the level of cooperating with something more complex like route / state management, could easily help with the part of the website, which will basically be NavigatorRails + Column> Text.

**Comment by Nipodemos** [comment]
> We can use Get's internationalization to change the language, and dynamic routes to access content directly via the link.
> The language can be saved using GetStorage so that if the person visits it again, they can read everything in their preferred language. A little ambitious, but I think the right thing to do.
> I would like to have a better communication channel too, currently there are groups spread across the telegram, discord, gitter, Facebook, but nothing officially supported.
that will be so wonderful 🤩 🤩 🤩 , can't wait to see it

For now, the changes made in documentation are ready, could you review and merge them?

**Comment by jonataslaw** [comment]
I am reviewing them right now, thank you very much for this contribution.

**Comment by ghprod** [comment]
> I have a domain purchased from getx.site that I used to validate the package on pub.dev, and some servers stopped (I pay some servers annually to do tests) I am thinking of doing the Get documentation, using Get with Flutter web, and put on that site. Perhaps it is the best option of all. More laborious, to be sure, but certainly the best option of all.

I agreed with this, so the docs itself can be used as a working example of GetX :+1:

**Comment by maginkgo** [comment]
See https://antora.org, and to use asciidoc

**Comment by Nipodemos** [comment]
@maginkgo by using this site wouldn't we have to convert markdown do ascii? that doesn't sound nice

**Comment by maginkgo** [comment]
Do not confuse ascii with asciidoc.  It is very similar to markdown but more powerful, and still Github compatible. https://asciidoctor.org/docs/asciidoc-vs-markdown/

**Comment by Nipodemos** [comment]
@maginkgo i looked in that asciidoc that you've mentioned, and i am amazed 😲 
absolutely loved it, and i want to convert all docs to asciidoc hahaha

that's only one problem... there is not flutter package to convert asciidoc to flutter 😢 😞 
so in order for we to continue on the idea of creating a flutter website of the docs, we need to keep markdown... which is a shame

if you find some pratical way of show asciidoc on flutter, let me know! i haven't found anything

**Comment by Nipodemos** [comment]
@jonataslaw did something happened while reviewing this PR? if you want some changes, just tell me 👍

**Comment by yang-lile** [comment]
> that's only one problem... there is not flutter package to convert asciidoc to flutter 😢 😞 so in order for we to continue on the idea of creating a flutter website of the docs, we need to keep markdown... which is a shame

Why be ashamed, usability is the first priority. The current getx documentation is like a piece of shit. Lacking a directory that is easy to navigate like Bloc, and no recommended best practices. Even so, if you still think about using flutter to create a document yourself, you are simply focusing on one thing and losing the other. Is GetStorge really comparable to commonly used databases? Where is the documentation for GetRouterOutlet? Trying to do everything will only result in doing nothing well.
